content/design-systems-for-developers/react/en/test.md

```diff
@@ -61,142 +61,149 @@ In the last chapter, our teammate did not want a red border around the `Button`'
 
 Undo the changes and commit again to pass your visual tests again.
 
-## Unit test functionality
+## Interaction tests
 
-Unit tests verify whether the UI code returns the correct output given a controlled input. They live alongside the component and help you validate specific functionality.
+So far, we've seen how visual testing enabled us to spot-check appearance and catch UI regressions. But as we continue to develop our design system, our components will eventually become responsible for more than just rendering the UI. At some point, they'll handle state management or even fetch data. That's where testing component interactions will help us.
 
-Everything is a component in modern view layers like React, Vue, and Angular. Components encapsulate diverse functionality from modest buttons to elaborate date pickers. The more intricate a component, the trickier it becomes to capture nuances using visual testing alone. That’s why we need unit tests.
+Interaction testing is a well-known pattern for verifying user behavior. You start by providing mocked data to set up your test, simulate user interactions with a testing library, and verify the UI changes. In Storybook, this happens in the browser, making it easier to debug failures because you're running tests in the same environment as you develop components: the browser.
 
-![Unit test components](/design-systems-for-developers/component-unit-testing.gif)
+To enable it, we're going to rely on Storybook's [`play`](https://storybook.js.org/docs/react/writing-stories/play-function) function and instrumented testing libraries to set up our tests and then use the [test-runner](https://storybook.js.org/docs/react/writing-tests/test-runner) verify our that the component renders correctly.
 
-For instance, our Link component is a little complicated when combined with systems that generate link URLs (“LinkWrappers” in ReactRouter, Gatsby, or Next.js). A mistake in the implementation can lead to links without a valid href value.
+### Setup the test runner
 
-Visually, it isn’t possible to see if the `href` attribute is there and points to the correct location, which is why a unit test can be appropriate to avoid regressions.
+Start by adding the necessary dependencies with:
 
-#### Unit testing hrefs
+```shell
+yarn add --dev jest@27 @storybook/jest @storybook/test-runner
```

Comment:
@jonniebigodes is the interactions addon pre-installed?

Reply:
@winkerVSbecks yes, that is correct. By default, new Storybook installations already come with the `@storybook/addon-interactions` already installed.